### PR TITLE
#300  Test drone

### LINKED ISSUE #300: Fix CI problems. stb2
CI does not launch

### LINKED ISSUE #314: Update prices command broken
Command says "all's good", but don't create new prices. Leaves only stale files

### PR DESCRIPTION
**Pull request by duker33**
Closes #300 
Closes #314

### CODE REVIEW

**Comment by duker33** [comment]
launched selenium. Since version `3.11.0` it forces only https connections to webdriver

**Comment by duker33** [comment]
now got problem with CI itself: it's cert expired, as it seems. Dig into letsencrypt ...

**Comment by duker33** [comment]
resolved with ssl cert

File: shopelectro/tests/helpers.py [hunk 1]
```diff
@@ -1,11 +1,15 @@
 import base64
 from contextlib import contextmanager
+import re
```
Comment (artemiy312) [inline shopelectro/tests/helpers.py]: `import` group should be before `from ... import ...`, isn't it?

File: shopelectro/tests/helpers.py [hunk 2]
```diff
@@ -82,3 +86,36 @@ def screen_fail(self, filename=''):
                 with open(f'screen__{filename}.png', 'wb') as f:
                     f.write(base64.b64decode(screen_b64.encode('ascii')))
             raise e
+
+    # @todo #300:60m Use or create clear selenium.WebElement behaviour system.from
+    #  Currently we return element from click method.
+    #  It's not clear design decision. Use or create your own behaviour system.
+    #  WebElement pipeline or smth.
```
Comment (artemiy312) [inline shopelectro/tests/helpers.py]: Maybe  [page objects pattern ](http://selenium-python.readthedocs.io/page-objects.html) will be helpful

File: .drone.yml [hunk 3]
```diff
@@ -44,7 +44,7 @@ pipeline:
       - python manage.py excel
       - python manage.py price
       - python manage.py collectstatic --noinput
-      - python manage.py test -v 3 --parallel --liveserver=test:8020-8030
+      - python manage.py test -v 3 --liveserver=test:8020-8030 --parallel=2
```
[the remark this answers is not in the document]
Comment (duker33) [reply .drone.yml]: @artemiy312 , i tried just to experiment with it. I'll return it to `--parallel` again

TODO

File: .drone.yml [hunk 4]
```diff
@@ -145,9 +145,11 @@ services:
       - RABBITMQ_DEFAULT_PASS=test
 
   selenium:
-    image: selenium/standalone-chrome-debug:3.3.0
+    image: selenium/standalone-chrome-debug:3.10.0
     environment: # https://github.com/SeleniumHQ/docker-selenium/issues/392
       - DBUS_SESSION_BUS_ADDRESS=/dev/null
+      - SCREEN_WIDTH=1366
```
[the remark this answers is not in the document]
Comment (duker33) [reply .drone.yml]: @artemiy312 , y, saw this comment. Let's discuss it there

File: docker/docker-compose.yml [hunk 5]
```diff
@@ -35,6 +35,8 @@ services:
     image: fidals/se-nodejs:dev
     volumes_from:
       - app
+    volumes:
+      - ./front/build:$SRC_DIR/front/build
```
[the remark this answers is not in the document]
Comment (duker33) [reply docker/docker-compose.yml]: @artemiy312 , nodejs should leave `./frount/build` files on host machine after build was finished.
Now it don't leave this files, that's why `dc run --rm nodejs gulp build` has no effect.

So, this code resolves bug #313

File: shopelectro/tests/helpers.py [hunk 6]
```diff
@@ -82,3 +86,36 @@ def screen_fail(self, filename=''):
...
+        dom_element.click()
+        return dom_element
+
+    def send_keys_and_wait(self, keys: str, locator, wait=True):
+        """
+        Safely send keys to element with given locator.
+
+        :param keys: should consist of letters, digits and spaces. No special symbols
+        """
+        assert re.match(r'[\w\d\s\-_]+', keys, re.I | re.U), \
```
[the remark this answers is not in the document]
Comment (duker33) [reply shopelectro/tests/helpers.py]: @artemiy312 , for example `backspace` characters don't appear in input field. That's why we get timeout exception for `send(Keys.BACKSPACE*n)`

File: shopelectro/tests/helpers.py (hunk 1 above)
Comment (duker33) [reply shopelectro/tests/helpers.py]: @artemiy312 , i was suprised by myself too, but at cryptobot project coala checked only lib names, without keywords. pep8 don't contain concrete recomendations about import/from order too.

I'll check linter again
TODO

File: shopelectro/tests/helpers.py (hunk 2 above)
Comment (duker33) [reply shopelectro/tests/helpers.py]: @artemiy312 , here we have problem not in test organizing in whole, but just in actions chaining.
So, i don't see this pattern helpful for this situation. But i'll add it to task desc, maybe implementor guy will find it helpful

TODO

File: docker/docker-compose.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (artemiy312) [reply docker/docker-compose.yml]: @duker33 
I tried to [reproduce](https://gist.github.com/artemiy312/5dd0dbea798f9349ff107f12775ef1b5) the bug and it works as a clock with `volumes_from: app`, therefore the `volumes:` option is redundant

File: shopelectro/tests/helpers.py (hunk 1 above)
Comment (artemiy312) [reply shopelectro/tests/helpers.py]: >pep8 don't contain concrete recomendations about import/from order too.

as I remember we decided to stick such ordering rules:
The first block is std-lib
The second is third-party
The third is local packages
The fourth is refarm-site packages

Each of group is ordered by import type and alphabetic. `import ...` blocks are higher than `from ... import ...`

Comment (duker33) [reply shopelectro/tests/helpers.py]: y, i'll fix it because of current rules
TODO

**Comment by duker33** [comment]
this discussion is the last one https://github.com/fidals/shopelectro/pull/301#discussion_r192407331
Move it to issue #313

File: docker/docker-compose.yml [hunk 7]
```diff
@@ -73,28 +75,32 @@ services:
...
+  # @todo #300:120m Resurrect coala.
+  #
   lint:
     image: fidals/coala-ci
     env_file:
       - env_files/paths
     working_dir: $SRC_DIR
-    volumes_from:
-      - app
+#    volumes:
```
Comment (artemiy312) [inline docker/docker-compose.yml]: Seems you forgot about it

File: docker/docker-compose.yml [hunk 8]
```diff
@@ -73,28 +75,32 @@ services:
...
       - SCREEN_HEIGHT=768
     networks:
       - se-backend
     # https://github.com/SeleniumHQ/docker-selenium#running-the-images
+    shm_size: 4G
     volumes:
       - /dev/shm:/dev/shm
 
+  # @todo #300:120m Resurrect coala.
+  #
```
Comment (artemiy312) [inline docker/docker-compose.yml]: It is redundant "#"

File: docker/docker-compose.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (artemiy312) [reply docker/docker-compose.yml]: Seems, you forgot about it

[the remark this answers is not in the document]
Comment (duker33) [reply docker/docker-compose.yml]: did no fixes throw last time, sorry. Was busy by burning #319

**Pull request merged**